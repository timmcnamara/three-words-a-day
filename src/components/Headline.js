import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { dark } from "../styles/styleGuide";

const HeadlineStyle = styled.h1`
  color: ${dark};
  font-size: 2em;
`;

const Headline = ({ title }) => {
  return <HeadlineStyle>{title}</HeadlineStyle>;
};

Headline.propTypes = {
  title: PropTypes.string
};

export default Headline;
