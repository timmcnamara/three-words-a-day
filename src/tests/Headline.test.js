import { render, cleanup } from "react-testing-library";
import React from "react";
import Headline from "../components/Headline";
import "jest-dom/extend-expect";

afterEach(cleanup);

it("renders headline component", () => {
  const { asFragment } = render(<Headline text="hello" />);

  // run a snapshot
  expect(asFragment()).toMatchSnapshot();
});

it("renders headline text", () => {
  const { getByText } = render(<Headline title="hello" />);
  expect(getByText("hello")).toBeInTheDocument();
});
