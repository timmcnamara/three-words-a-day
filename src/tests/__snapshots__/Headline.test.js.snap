// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders headline component 1`] = `
<DocumentFragment>
  <h1
    class="sc-bdVaJa cFuNTo"
  />
</DocumentFragment>
`;
